test/storage/bplustree_test.cpp

```diff
@@ -0,0 +1,1468 @@
+  // from right side without putting anything in the right side
+  // Hence the size may be 31 at some points if we used 64.
+  bplustree->SetInnerNodeSizeUpperThreshold(16);
+  bplustree->SetLeafNodeSizeUpperThreshold(16);
+  bplustree->SetInnerNodeSizeLowerThreshold(6);
+  bplustree->SetLeafNodeSizeLowerThreshold(6);
+  std::set<int> keys;
+  for (unsigned i = 0; i < 1000; i++) {
+    BPlusTree<int, TupleSlot>::KeyElementPair p1;
+    int k = rand_r(&globalseed) % 5000;
+    while (keys.find(k) != keys.end()) k++;
+    keys.insert(k);
+    p1.first = k;
+    bplustree->Insert(p1, predicate);
+
+    auto keys_copy = keys;
+
+    // Structural Integrity Verification Everytime
+    EXPECT_EQ(bplustree->StructuralIntegrityVerification(*keys_copy.begin(), *keys_copy.rbegin(), &keys_copy,
+                                                         bplustree->GetRoot()),
+              true);
+    EXPECT_EQ(keys_copy.size(), 0);
+  }
+
+  // Delete All keys except one - As root empty is not handled by delete yet
+  for (int i = 0; i < 999; i++) {
+    auto iter = keys.rbegin();
+    BPlusTree<int, TupleSlot>::KeyElementPair p1;
+    p1.first = *iter;
+    keys.erase(*iter);
+    EXPECT_EQ(bplustree->DeleteElement(p1), true);
+    std::set<int> newkeys = keys;
+
+    // Structural Integrity Test Everytime
+    EXPECT_EQ(
+        bplustree->StructuralIntegrityVerification(*newkeys.begin(), *newkeys.rbegin(), &newkeys, bplustree->GetRoot()),
+        true);
+    EXPECT_EQ(newkeys.size(), 0);
+  }
+
+  // Insert Again
+  for (unsigned i = 0; i < 1000; i++) {
+    BPlusTree<int, TupleSlot>::KeyElementPair p1;
+    int k = rand_r(&globalseed) % 5000;
+    while (keys.find(k) != keys.end()) k++;
+    keys.insert(k);
+    p1.first = k;
+    bplustree->Insert(p1, predicate);
+
+    auto keys_copy = keys;
+
+    // Structural Integrity Verification Everytime
+    EXPECT_EQ(bplustree->StructuralIntegrityVerification(*keys_copy.begin(), *keys_copy.rbegin(), &keys_copy,
+                                                         bplustree->GetRoot()),
+              true);
+    EXPECT_EQ(keys_copy.size(), 0);
+  }
+
+  // Delete Again now two keys remaining
+  for (int i = 0; i < 999; i++) {
+    auto iter = keys.rbegin();
+    BPlusTree<int, TupleSlot>::KeyElementPair p1;
+    p1.first = *iter;
+    keys.erase(*iter);
+    EXPECT_EQ(bplustree->DeleteElement(p1), true);
+    std::set<int> newkeys = keys;
+
+    // Structural Integrity Test Everytime
+    EXPECT_EQ(
+        bplustree->StructuralIntegrityVerification(*newkeys.begin(), *newkeys.rbegin(), &newkeys, bplustree->GetRoot()),
+        true);
+    EXPECT_EQ(newkeys.size(), 0);
+  }
+
+  // Check Both still present
+  auto iter = keys.begin();
+  for (unsigned i = 0; i < 2; i++) {
+    EXPECT_EQ(bplustree->IsPresent(*iter), true);
+    iter++;
+  }
+
+  // Free Everything
+
+  delete bplustree;
+}
+
+void StructuralIntegrityTestWithRandomInsertAndDelete2() {
```

Comment:
Can we name it in a more meaningful way instead of just adding 2 at the end?

Reply:
Removed these 2 tests since they were redundant.